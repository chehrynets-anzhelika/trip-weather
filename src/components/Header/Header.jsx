import React from 'react';
import AddTrip from '../AddTrip/AddTrip';
import Sort from '../Sort/Sort';
import  styles  from './header.module.css';


const Header = () => {
    return (
        <>
        <header className={styles.header}>
        <div className='container'>
        <nav className={styles.nav}>
             <a href='/' className={styles.logo} onClick={(e) => e.preventDefault()}>Weather Forecast</a>
             <div className={styles.navBtnContainer}>
                <Sort/>
                <AddTrip/>
             </div>
        </nav>
        </div>
        </header>
        </>
        
    );
}

export default Header;
