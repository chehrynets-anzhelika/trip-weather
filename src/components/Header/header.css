.header {
    color: #FFF;
    position: fixed;
    width: 100%;
    top: 20px;
}

.header-nav {
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 100%;
}

.header-logo {
   font-size: 36px;
   text-decoration: none;
   color: #FFF;
}