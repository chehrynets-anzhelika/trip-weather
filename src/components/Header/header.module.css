.header {
    color: #FFF;
    position: sticky;
    width: 100%;
    top: 20px;
    z-index: 4;
}

.nav {
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 100%;
}

.logo {
   font-size: 36px;
   text-decoration: none;
   color: #FFF;
}

.navBtnContainer {
    display: flex;
}

@media screen and (max-width: 640px) {
    .logo {
        font-size: 24px;
     }
     .nav {
        flex-direction: column;
        gap: 15px;
    }
}

