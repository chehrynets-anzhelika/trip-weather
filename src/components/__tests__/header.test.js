import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from "../Header/Header";
import * as hooks from "react-redux";


jest.mock("react-redux");
const mockedDispatch = jest.spyOn(hooks, "useDispatch");


test("Schould render component Header", () => {
    mockedDispatch.mockReturnValue(jest.fn());
    render(<Header />);
    expect(screen.getByTestId("header")).toMatchSnapshot();
})