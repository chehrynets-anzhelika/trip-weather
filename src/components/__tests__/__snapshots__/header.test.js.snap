// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schould render component Header 1`] = `
<header
  class="header"
  data-testid="header"
>
  <div
    class="container"
  >
    <nav
      class="nav"
    >
      <a
        class="logo"
        href="/"
      >
        What's Weather?
      </a>
      <div
        class="navBtnContainer"
      >
        <div>
          <div
            class="sortContainer"
            data-testid="sort"
          >
            <button
              class="header-btn btn"
            >
              <span
                class="content"
              >
                Sort
              </span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-caret-down btnArrow"
                data-icon="caret-down"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
        <div>
          <button
            class="header-btn btn"
          >
            <span>
              Login
            </span>
          </button>
        </div>
        <button
          class="header-btn addTripBtn btn"
          data-testid="button"
        >
          Add trip
        </button>
      </div>
    </nav>
  </div>
</header>
`;
